### PR TITLE
Moths can now eat clothes

### PR DESCRIPTION
**Pull request by challade**
Moths can now eat certain clothes.

[Changelogs]: # (Your PR should contain a detailed changelog of notable changes, titled and categorized appropriately. This includes, new features, sprites, sounds, balance changes, admin tools, map edits, removals, big refactors, config changes, hosting changes and important fixes. An example changelog has been provided below for you to edit. If you need additional help, read https://github.com/tgstation/tgstation/wiki/Changelogs)

:cl: Challade
Moths can now refresh themselves on the clothing off their back, and of their generous companions. Provides nutrition and fury from nude stationmates.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

### CODE REVIEW

**Comment by ghost** [comment]
as much as races should be cosmetic, the mental image of a moth eating the clothes right of a wizard is far too much

**Comment by Pyko1** [comment]
we better make moths vulnerable to pest spray and flyswatters if we add this.

Also make hotspots suck them in like a singularity too.

**Comment by JohnGinnane** [comment]
I want this very much

**Comment by yoyobatty** [comment]
Since I technically co-authored, it will need a code check from someone else.

**Comment by challade** [comment]
>co-authored
more like actually made it work, I was like a monkey fucking around with the tab button, thanks Yoyo.

File: hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm [hunk 1]
```diff
@@ -37,6 +37,23 @@
...
+
+/mob/living/carbon/human/attackby(obj/item/W, mob/living/carbon/human/user)
+	if(dna.species.id == "moth") //specifies the species
+		var/static/list/item_types = list(/obj/item/clothing,
+		/obj/item/clothing/neck,
+		/obj/item/clothing/head,
+		/obj/item/clothing/mask,
+		/obj/item/clothing/under,
+		/obj/item/clothing/shoes) //lists the tasty snacks
+		if(is_type_in_list(W, /obj/item/clothing))
```
Comment (yoyobatty) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm]: `if(is_type_in_list(W, /obj/item/clothing))`
should be `if(is_type_in_list(W, item_types))`

File: hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm [hunk 2]
```diff
@@ -37,6 +37,23 @@
...
+		/obj/item/clothing/neck,
+		/obj/item/clothing/head,
+		/obj/item/clothing/mask,
+		/obj/item/clothing/under,
+		/obj/item/clothing/shoes) //lists the tasty snacks
+		if(is_type_in_list(W, item_types))
+			var/obj/item/clothing/C = W
+			playsound(get_turf(src), 'sound/items/eatfood.ogg', 70,1)
+			visible_message("<span class='alert'>[user] bites into a [C].</span>")
+			nutrition += 20
```
Comment (Jujumatic) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm]: god damn now that's what i call nutriment, you'll be a fat shit in no time lmao

File: hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm [hunk 3]
```diff
@@ -37,6 +37,23 @@
...
+			C.take_damage(50, BRUTE, "melee", 1)
```
Comment (Jujumatic) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm]: ow :(

**Review by Jujumatic — DISMISSED** [review]
this is the kind of content hippiestation needs in its darkest hours

File: hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm (hunk 2 above)
Comment (JohnGinnane) [reply hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm]: yeah might want to lower this, what's the nutrition level for becoming fat?

File: hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm (hunk 3 above)
Comment (yoyobatty) [reply hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm]: 2 or 3 is probably more respectable

**Comment by Carbonhell** [comment]
I am against any race benefit
But this is way too powerful, I have to +1 it

**Comment by DaydreamPuppeh** [comment]
I wouldn't even call this a benefit because of how fast it would get you fat.

**Comment by Carbonhell** [comment]
https://github.com/tgstation/tgstation/pull/39350

**Comment by Mothblocks** [comment]
@Carbonhell dude what i didn't even notice this

**Comment by Carbonhell** [comment]
hm?no problems, i was just saying that if the tg one is merged we don't need ours
although since we have a separate mothmen.dm file we may need this anyway

**Comment by Mothblocks** [comment]
I know there's no problems, I just thought it was funny and thought my PR was mirrored immediately because the titles were the same.

**Comment by challade** [comment]
Update: I changed the damage clothes take to 3, and the nutrients gained to 10. I can't seem to find how much it takes for the clothing to dissipate from a certain amount of brute damage. So far I'm gonna keep it that way for now. I'm hoping someone mentions it here because I was hoping that it'd be a light 3 bite snack, for 3 reasons:
>1. It'd be too powerful if a moth can just walk around lavaland depowered station with an extra jumpsuit in their bag to munch on while their human companions starve.
>2. I want all the moths to eat all the jumpsuits on the station and create a clothes shortage where they resort to mad max jumping assistants in the halls, dragging them into maint, stripping them nude, and nibbling their clothing in front of them.
>3. this connects to 2, cargo usually fucks off, they'd appreciate if there was a clothes shortage, to order some snazzy clothing for the nude and helpless.

**Comment by JohnGinnane** [comment]
Looks like jumpsuits have health of 200:

image

**Comment by challade** [comment]
Update 2: Bumped c.damage back to 50, that way it will take 4 bites to completely strip a young jumpsuit of its flesh and nutrients.

**Comment by Jujumatic** [comment]
hey this means you can eat the clown's shoes as well

+9999999 sudden realisation

**Comment by challade** [comment]
Shoes, mask, gloves, jumpsuit, clown. You can eat the whole thing, except maybe the clown, since moths get disgust from meat.

**Comment by Manly-Dwarf** [comment]
If this gets merged, I will port things from Atlas that should have never resurfaced.

**Comment by JohnGinnane** [comment]
is there something wrong with this @Kawaii-Big-

**Comment by Manly-Dwarf** [comment]
Not sure. I think that races should be mostly cosmetic, with only minute differences here and there. Or in other words, we can't have our hardmode plasmaman races because of the cosmetic rule, so I now rebel against anything that may be fun for races.
In all seriousness though if this gets merged I'll probably port a thing or two over from Atlas, maybe make it so catbeast scum can't roll antag at all or be converted so people can play as them again if they feel like dying horrible deaths.

**Comment by JohnGinnane** [comment]
@Kawaii-Big-Boss you can port that stuff now, you don't need to wait for this, the idea behind "strictly cosmetic" is to curb any powergaming

File: hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm [hunk 4]
```diff
@@ -37,6 +37,23 @@
...
 		H.dna.features["moth_wings"] = "Burnt Off"
 		handle_mutant_bodyparts(H)
-		
+
 /datum/species/moth/check_roundstart_eligible()
 	return TRUE
+
+/mob/living/carbon/human/attackby(obj/item/W, mob/living/carbon/human/user)
+	if(dna.species.id == "moth") //specifies the species
+		var/static/list/item_types = list(/obj/item/clothing,
```
Comment (Carbonhell) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/mothmen.dm:46]: Why is this a static list?I don't think this has any effect inside a proc. Also isn't there any hook for species in human attackby already that you could use? And couldn't you just use subtypesof(/obj/item/clothing)? I don't think is_type_in_list will work with this considering it'll do a direct check type==type if i'm not wrong.

**Comment by ghost** [comment]
-1

**Comment by ghost** [comment]
im torn
it'd be really funny to have moths just eat up your fuckin' clothes
but i also don't want any snowflakiness between species
i'll do a coin flip

tails, ok it's gotta be -1 unfortunately

also kbb you can still port those things if you'd like

**Comment by Carbonhell** [comment]
What bruce said, I am very sorry man, gotta deny this

**Comment by Jujumatic** [comment]
fuck it just make everyone able to eat clothes then i want this feature so badly 😢

**Pull request closed without merging**